### ISSUE TITLE
Issue with CORS only on Firefox

### ISSUE THREAD

**Issue by MystPhysX** [susanBuck/e28-fall20#73]
Hello,

I am having an issue with the api on only Firefox for some reason. When trying to get any auth protected resources it works well the first time and then times out every subsequent time and complains about missing CORS headers like follows:

![image](https://user-images.githubusercontent.com/6554675/101989995-029f8080-3cca-11eb-9f0d-9e19c11ee16a.png)

If I leave it alone for a while it'll work again the first time and then stop again for some reason.

However, none of this happens in Chrome. I have yet to come across any issues in Chrome at all.

Any ideas? I am using Nginx instead of Apache, but Nginx isn't doing anything to the headers and Chrome would complain if it was.

**Comment by R-Elijah** [comment]
Opened this on my personal account by mistake. The URL is http://e28p3.mystphy.sx/ for the site.

**Comment by susanBuck** [comment]
Hi @R-Elijah -

Doing a little bit of research on this - have not come to a conclusion yet, but wanted to share some of my notes.

One, here's the responses when the requests are **successful**:
![image](https://user-images.githubusercontent.com/237711/102018168-788f0080-3d39-11eb-89c1-0d79ce0337ee.png)
![image](https://user-images.githubusercontent.com/237711/102018616-4e8b0d80-3d3c-11eb-8044-c7f21f906969.png)

And here's the response when the requests **fail**:
![image](https://user-images.githubusercontent.com/237711/102018588-2ac7c780-3d3c-11eb-8e30-f64d676da4ea.png)

It looks like the requests are returning a 500 server error, and as a result of that we don't have the correct CORS headers, which is why we see the CORS error in the browser.

Which is to say, I don't think the root of the problem is the CORS, but instead the fact that requests are returning a 500 server error. 

Where are you hosting this? Any chance there are any sort of rate limits on the server? 

Your app does make several requests to the API when mounted - I'm wondering if it's triggering some rate limit, which would explain why if you give it a little time, it works again.

**Comment by susanBuck** [comment]
Follow-up thought: The fact that it works on Chrome but not Firefox contradicts my rate-limit theory. 

I have to run out now, but will look more into this this evening or Monday.

**Comment by MystPhysX** [comment]
It's hosted on GCP and I don't believe there should be any rate limit. I'm still trying to figure it out but I'll probably move it over to a DigitalOcean instance if I cannot figure it out by tomorrow and see what happens there.

**Comment by R-Elijah** [comment]
It's hosted on GCP and I don't believe there's a rate limit on the instance. I will try moving it over to DigitalOcean later today and see if it happens there.

**Comment by susanBuck** [comment]
Cool - I'll also launch something to GCP to see if I can recreate on my end later/Monday!

**Comment by R-Elijah** [comment]
I looked into the response a bit more and found that the 500 internal server error was coming from Cloudflare and not my server. I disabled Cloudflare proxying for just the API and now it works great on Firefox as well.

Not sure what exactly the issue was that was only preventing FF from working but disabling proxying isn't a very big thing so I'll stick with that.

Thank you very much @susanBuck and I want to apologise for taking up your time for such a strange one off bug.
